internal/auth/auth.go

```diff
@@ -344,3 +352,68 @@ func persistConfiguration() error {
 	_ = os.MkdirAll(filepath.Dir(path), os.ModePerm)
 	return os.WriteFile(path, data, 0o600)
 }
+
+func RefreshCloudSession() error {
+	if globalConfig == nil {
+		c := DefaultConfig()
+		globalConfig = &c
+	}
+
+	data := url.Values{}
+	data.Set("refresh_token", globalConfig.CloudRefreshToken)
+	data.Set("client_id", cloudIdentityServiceClientId)
+	data.Set("grant_type", "refresh_token")
+
+	req, err := http.NewRequest("POST", cloudIdentityServiceTokenUrl, strings.NewReader(data.Encode()))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send post request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var response *api.AccessToken
+
+	err = json.NewDecoder(resp.Body).Decode(&response)
+	if err != nil {
+		return fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	globalConfig.CloudAccessToken = response.Token
+	globalConfig.CloudRefreshToken = response.RefreshToken
+
+	return persistConfiguration()
+}
+
+func IsAccessTokenExpired() (bool, error) {
+	if globalConfig == nil {
+		c := DefaultConfig()
+		globalConfig = &c
+	}
+
+	claims := jwt.MapClaims{}
+	_, _, err := jwt.NewParser().ParseUnverified(globalConfig.CloudAccessToken, claims)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse jwt token: %w", err)
+	}
+
+	exp, ok := claims["exp"].(float64)
+	if !ok {
+		return false, fmt.Errorf("invalid jwt token")
+	}
+
+	return time.Now().Unix() > int64(exp), nil
+}
+
+func ShouldCheckAccessTokenExpiry() bool {
+	if globalConfig == nil {
```

Comment:
The function `ShouldCheckAccessTokenExpiry` seems like it is meant only to make a decision and not to update global state. This is an unexpected side-effect which should be avoided.

If globalConfig is nil, just return `true` so that access token expiry check goes through. No need to set `globalConfig`